separate xxxResultsetInstance types

mbaudis opened the pull request:
```
  resultSets:
    type: array
    items:
      oneOf:
        - $ref: '#/$defs/BooleanResultsetInstance'
        - $ref: '#/$defs/CountResultsetInstance'
        - $ref: '#/$defs/RecordsResultsetInstance'
```

This is an attempt to address the problem of Boolean and count resultSets discused in https://github.com/ga4gh-beacon/beacon-v2/discussions/222 . Should actually be non-breaking.

Also changes `info` to point to the prototype definition.

mbaudis committed:
separate xxxResultsetInstance types

```
  resultSets:
    type: array
    items:
      oneOf:
        - $ref: '#/$defs/BooleanResultsetInstance'
        - $ref: '#/$defs/CountResultsetInstance'
        - $ref: '#/$defs/RecordsResultsetInstance'
```

This is an attempt to address the problem of Boolean and count resultSets discused in https://github.com/ga4gh-beacon/beacon-v2/discussions/222 . Should actually be non-breaking.

Also changes `info` to point to the prototype definition.

costero-e reviewed (changes requested):
Indeed, this is a non-breaking change but the validator can't tell the difference between one of the different instances unless we add some `returnedGranularity` parameter, hence the proposal is still not valid. Here is the complaint from the validator:
 ```
Message:
JSON is valid against more than one schema from 'oneOf'. Valid schema indexes: 0, 1, 2.
Schema path:
#/properties/resultSets/items/oneOf
``` 
Our work here in the [resultSet proposal](https://github.com/ga4gh-beacon/beacon-v2/blob/resultSet-response_per-granularity/framework/json/responses/sections/beaconResultsets.json) had this addressed but then it's a breaking change.
The other solution is what we said in the Dev Scouts, where we agreed to just remove results and resultsCount from required.

mbaudis committed:
re-wrapping the resultSets typed definitions

... test since validator failed on items: oneOf construct

mbaudis commented:
@costero-e Can you check the updated version please? Since oneOf now isn't for list items (same as in the variants model ...).

costero-e commented:
It keeps complaining. The problem is that without any specific parameter for each of the instances of the `oneOf`, then, validator says that `oneOf` is not able to say what belongs to each instance, as a BooleanInstance could be also a CountInstance or a Record Instance. RecordInstance with results is obvious that will be a RecordInstance but the other ones are the ones conflicting.

mbaudis commented:
@costero-e Well, I'm happy with just going the road of having the status quo w/ removal of the `required` for `resultsCount` and `results`.

Alternatively: Would `type` parameter with a constant (as e.g. VRS is doing it...) help w/ validation? 

```
RecordsResultsetInstance:
    type: object
    properties:
        type:
            type: string
            const: records
...
```
or
```
RecordsResultsetInstance:
    type: object
    properties:
        resultsetGranularity:
            type: string
            const: record
...
```

costero-e commented:
Yes, this would absolutely solve the problem but would be breaking change. I thought from the Dev Scouts meeting we had last week that we decided to just remove `resultsCount` and `results` from required though, as we understood that resultSets response is a `record` response but that an owner can decide to give the detail of the record response depending on the permissions. In fact, currently, we can return a record response without records in`results`. I haven't talked to @jrambla about this anyways, I'm just giving my opinion.

mbaudis commented:
@costero-e Yeah, I had commented in that direction - one of the 2 options https://github.com/ga4gh-beacon/beacon-v2/discussions/222#discussioncomment-13077549 This one (if made working) is the more expressive & controlled one - we just can then pick which is the preferred one. I'll update this and do the alternative.

mbaudis committed:
adding resultsetGranularity

... as per discussion https://github.com/ga4gh-beacon/beacon-v2/pull/230#issuecomment-2871955061

gsfk reviewed (commented):
I left some comments on #231, which I think I prefer for the moment.

costero-e commented:
Closing this PR @mbaudis as looks like #231 is the one that has more acceptance now.
